src/main/java/com/minecolonies/coremod/commands/ActionArgument.java

```diff
@@ -0,0 +1,89 @@
+package com.minecolonies.coremod.commands;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+import org.jetbrains.annotations.NotNull;
+
+import com.minecolonies.coremod.commands.ActionArgumentType.Is;
+
+public class ActionArgument
+{
+    @NotNull private final String name;
+    @NotNull private final ActionArgumentType type;
+    private final boolean required;
+    private boolean valueIsSet = false;
+    @Nullable private Object value;
+    @Nullable private final List<ActionArgument> actionArgumentList;
+
+    public ActionArgument(@NotNull final String name, @NotNull final ActionArgumentType type, final Is optionalOrRequired, @Nullable final ActionArgument ... actionArguments
+)
+    {
+        super();
+        this.name = name;
+        this.type = type;
+        this.required = optionalOrRequired == Is.REQUIRED;
+        this.actionArgumentList = Arrays.asList(actionArguments);
+    }
+
+    public String getName()
+    {
+        return name;
+    }
+
+    public ActionArgumentType getType()
+    {
+        return type;
+    }
+
+    public boolean isRequired()
+    {
+        return required;
+    }
+
+    public List<ActionArgument> getActionArgumentList()
+    {
+        return actionArgumentList;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Return a copy of "actionArgumentList". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2384)